flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotProvider.java

```diff
@@ -42,6 +42,23 @@
  */
 public interface SlotProvider {
 
+	/**
+	 * Allocating slot with specific requirement.
+	 *
+	 * @param slotRequestId identifying the slot request
+	 * @param scheduledUnit The task to allocate the slot for
+	 * @param slotProfile profile of the requested slot
+	 * @param allocationTimeout after which the allocation fails with a timeout exception
+	 * @return The future of the allocation
+	 */
+	default CompletableFuture<LogicalSlot> allocateSlot(
```

Comment:
Seems this interface is only used by tests. Do we really need it or can we make the tests to invoke the underlying interface directly?

Reply:
I think de facto there is only queued scheduling so the other method should be removed. However, I am reverting my change because it's not needed for the scheduler.